src/main/kotlin/org/kamiblue/client/module/modules/player/AutoEat.kt

```diff
@@ -143,25 +173,32 @@ internal object AutoEat : Module(
      * @return `true` if food found and moved
      */
     private fun SafeClientEvent.moveFoodToHotbar(): Boolean {
-        val slotFrom = player.storageSlots.firstItem<ItemFood, Slot> {
-            isValid(it)
-        } ?: return false
+        val slotFrom = getSlotOfItemFood(prefferedLevel, player.storageSlots) ?: return false
 
         moveToHotbar(slotFrom) {
             val item = it.item
             item !is ItemTool && item !is ItemBlock
         }
-
         return true
     }
 
-    private fun SafeClientEvent.isValid(itemStack: ItemStack): Boolean {
+    fun SafeClientEvent.getSlotOfItemFood(prefferedLevel: PreferedLevel, inventory: List<Slot>): Slot? {
+        return inventory.firstItem<ItemFood, Slot> {
+            isValid(it, AutoEat.prefferedLevel)
+        } ?: when (prefferedLevel) {
+            PreferedLevel.NORMAL -> null
+            PreferedLevel.GAP -> getSlotOfItemFood(PreferedLevel.NORMAL, inventory)
+            else -> getSlotOfItemFood(PreferedLevel.GAP, inventory)
+        }
+    }
+
+    private fun SafeClientEvent.isValid(itemStack: ItemStack, prefferedLevel: PreferedLevel): Boolean {
         val item = itemStack.item
+        if ((item !is ItemFood) || (item == Items.CHORUS_FRUIT) || (isBadFood(itemStack, item) and !eatBadFood)) {
+            return false
+        }
 
-        return item is ItemFood
-            && item != Items.CHORUS_FRUIT
-            && (eatBadFood || !isBadFood(itemStack, item))
-            && player.canEat(item == Items.GOLDEN_APPLE)
+        return prefferedLevel.isValid(item, itemStack)
     }
```

Comment:
I don't see isValid being used anywhere?? Am I missing something?

Reply:
It is, line 187, 88, 85.